[MSVC] Test_ONNX_layers.RandomNormalLike_basic/0 and Test_ONNX_layers.RandomNormalLike_complex/0 failed on MSVC

### System Information

Windows 11 64 bit
VS2022 17.14.32

### Detailed description

Hi All,

OpenCV failed to run 2 tests with MSVC on windows. It can be reproduced on 7c7ee9749317260ad8302e3ed4365bd614e2b018 commit on master branch. Could you please help look at this issue?

Test log: [Test16.log](https://github.com/user-attachments/files/28044957/Test16.log)

**Error info:**
```
C:\gitP\opencv\opencv\build_amd64\bin\Release>opencv_test_dnn.exe --gtest_filter="Test_ONNX_layers.RandomNormalLike_basic/0:Test_ONNX_layers.RandomNormalLike_complex/0"
CTEST_FULL_OUTPUT
OpenCV version: 4.14.0-pre
OpenCV VCS version: 4.13.0-277-g7c7ee97493-dirty
Build type: Debug Release
WARNING: build value differs from runtime: Release
Compiler: C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.44.35207/bin/Hostx64/x64/cl.exe  (ver 19.52.36414.16)
Algorithm hint: ALGO_HINT_ACCURATE
HAL: YES (ipp (ver 0.0.1))
Parallel framework: ms-concurrency (nthreads=16)
CPU features: SSE SSE2 SSE3 *SSE4.1 *SSE4.2 *AVX *FP16 *AVX2 *AVX512-SKX
Intel(R) IPP version: ippIP AVX-512F/CD/BW/DQ/VL (k0) 2026.0.0 (-) Mar 27 2026
Intel(R) IPP features code: 0x7300000
TEST: Skip tests with tags: 'mem_6gb', 'verylong', 'dnn_skip_opencv_backend', 'dnn_skip_cpu', 'dnn_skip_cpu_fp16', 'dnn_skip_ocl', 'dnn_skip_ocl_fp16', 'dnn_skip_onnx_conformance', 'dnn_skip_parser'
Note: Google Test filter = Test_ONNX_layers.RandomNormalLike_basic/0:Test_ONNX_layers.RandomNormalLike_complex/0
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from Test_ONNX_layers
[ RUN      ] Test_ONNX_layers.RandomNormalLike_basic/0, where GetParam() = OCV/CPU
[ERROR:0@0.034] global onnx_importer.cpp:1040 cv::dnn::dnn4_v20251223::ONNXImporter::handleNode DNN/ONNX: ERROR during processing node with 1 inputs and 1 outputs: [RandomNormalLike]:(onnx_node_output_0!y) from domain='ai.onnx'
unknown file: error: C++ exception with description "OpenCV(4.14.0-pre) C:\gitP\opencv\opencv\modules\dnn\src\onnx\onnx_importer.cpp:1059: error: (-2:Unspecified error) in function 'cv::dnn::dnn4_v20251223::ONNXImporter::handleNode'
> Node [RandomNormalLike@ai.onnx]:(onnx_node_output_0!y) parse error: OpenCV(4.14.0-pre) C:\gitP\opencv\opencv\modules\dnn\src\net_impl.hpp:108: error: (-2:Unspecified error) Can't create layer "onnx_node_output_0!y" of type "RandomNormalLike" in function 'cv::dnn::dnn4_v20251223::Net::Impl::getLayerInstance'
> " thrown in the test body.
[  FAILED  ] Test_ONNX_layers.RandomNormalLike_basic/0, where GetParam() = OCV/CPU (2 ms)
[ RUN      ] Test_ONNX_layers.RandomNormalLike_complex/0, where GetParam() = OCV/CPU
[ERROR:0@0.036] global onnx_importer.cpp:1040 cv::dnn::dnn4_v20251223::ONNXImporter::handleNode DNN/ONNX: ERROR during processing node with 1 inputs and 1 outputs: [RandomNormalLike]:(onnx_node_output_0!rnl_out) from domain='ai.onnx'
unknown file: error: C++ exception with description "OpenCV(4.14.0-pre) C:\gitP\opencv\opencv\modules\dnn\src\onnx\onnx_importer.cpp:1059: error: (-2:Unspecified error) in function 'cv::dnn::dnn4_v20251223::ONNXImporter::handleNode'
> Node [RandomNormalLike@ai.onnx]:(onnx_node_output_0!rnl_out) parse error: OpenCV(4.14.0-pre) C:\gitP\opencv\opencv\modules\dnn\src\net_impl.hpp:108: error: (-2:Unspecified error) Can't create layer "onnx_node_output_0!rnl_out" of type "RandomNormalLike" in function 'cv::dnn::dnn4_v20251223::Net::Impl::getLayerInstance'
> " thrown in the test body.
[  FAILED  ] Test_ONNX_layers.RandomNormalLike_complex/0, where GetParam() = OCV/CPU (1 ms)
[----------] 2 tests from Test_ONNX_layers (5 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test case ran. (6 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] Test_ONNX_layers.RandomNormalLike_basic/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.RandomNormalLike_complex/0, where GetParam() = OCV/CPU

 2 FAILED TESTS
```

### Steps to reproduce

```
1. Open VS2022 x64 Native Command Prompt tools
2. git clone https://github.com/opencv/opencv.git C:\gitP\opencv\opencv
3. git clone https://github.com/opencv/opencv_extra.git C:\gitP\opencv\opencv_extra
4. mkdir build_amd64 && cd /d build_amd64
5. cmake -G "Visual Studio 17 2022" -A x64 -DCMAKE_SYSTEM_VERSION=10.0.26100.0 -DWITH_IPP=ON -DBUILD_SHARED_LIBS=OFF -DBUILD_PERF_TESTS=ON -DBUILD_TESTS=ON -DBUILD_EXAMPLES=ON -DWITH_OPENCL=OFF -DBUILD_DOCS=OFF -DWITH_CUDA=OFF -DWITH_EIGEN=OFF ..
6. msbuild /m /p:Platform=x64 /p:Configuration=Release OpenCV.sln /t:Rebuild /p:BuildInParallel=true 
7. set OPENCV_TEST_DATA_PATH=C:\gitP\opencv\opencv_extra\testdata
8. cd /d C:\gitP\opencv\opencv\build_amd64\bin\Release
9. opencv_test_dnn.exe --gtest_filter="Test_ONNX_layers.RandomNormalLike_basic/0:Test_ONNX_layers.RandomNormalLike_complex/0"
```

### Issue submission checklist

- [x] I report the issue, it's not a question
- [x] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [x] I updated to the latest OpenCV version and the issue is still there
- [x] There is reproducer code and related data files (videos, images, onnx, etc)

--- a/modules/dnn/include/opencv2/dnn/all_layers.hpp
+++ b/modules/dnn/include/opencv2/dnn/all_layers.hpp
@@ -1204,6 +1204,12 @@ CV__DNN_INLINE_NS_BEGIN
         static Ptr<TopKLayer> create(const LayerParams& params);
     };
 
+    class CV_EXPORTS RandomNormalLikeLayer : public Layer
+    {
+    public:
+        static Ptr<RandomNormalLikeLayer> create(const LayerParams& params);
+    };
+
 //! @}
 //! @}
 CV__DNN_INLINE_NS_END

--- a/modules/dnn/src/init.cpp
+++ b/modules/dnn/src/init.cpp
@@ -200,6 +200,7 @@ void initializeLayerFactory()
     CV_DNN_REGISTER_LAYER_CLASS(ScatterND,      ScatterNDLayer);
     CV_DNN_REGISTER_LAYER_CLASS(Tile,           TileLayer);
     CV_DNN_REGISTER_LAYER_CLASS(TopK,           TopKLayer);
+    CV_DNN_REGISTER_LAYER_CLASS(RandomNormalLike, RandomNormalLikeLayer);
 
     CV_DNN_REGISTER_LAYER_CLASS(Quantize,         QuantizeLayer);
     CV_DNN_REGISTER_LAYER_CLASS(Dequantize,       DequantizeLayer);

--- a/modules/dnn/src/layers/randomnormallike_layer.cpp
+++ b/modules/dnn/src/layers/randomnormallike_layer.cpp
@@ -6,12 +6,11 @@
 
 #include "../precomp.hpp"
 #include "layers_common.hpp"
-#include <opencv2/dnn/layer.details.hpp>
 #include <cmath>
 
 namespace cv { namespace dnn {
 
-class RandomNormalLikeLayerImpl CV_FINAL : public Layer
+class RandomNormalLikeLayerImpl CV_FINAL : public RandomNormalLikeLayer
 {
 public:
     RandomNormalLikeLayerImpl(const LayerParams& params)
@@ -117,6 +116,9 @@ private:
     int depth;
 };
 
-CV_DNN_REGISTER_LAYER_CLASS_STATIC(RandomNormalLike, RandomNormalLikeLayerImpl);
+Ptr<RandomNormalLikeLayer> RandomNormalLikeLayer::create(const LayerParams& params)
+{
+    return makePtr<RandomNormalLikeLayerImpl>(params);
+}
 
 }} // namespace cv::dnn

